plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/CustomParquetWriter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.parquet;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.api.WriteSupport;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.schema.MessageType;
+
+import java.io.IOException;
+import java.util.List;
+
+public class CustomParquetWriter
```

Comment:
For testing please get rid of CustomParquetWriter and CustomWriteSupport and use a pre-existing file generated with the new encoding instead.
You can temporarily update `ParquetWriter#initColumnWriters` to use `.withByteStreamSplitEncoding(true)`. Then write a parquet file with a double and a float column type. That should give us a file with this encoding which we can use for reader tests.
We can skip testing in AbstractTestParquetReader and rely on standalone test reading from pre-existing file instead.